pydicom/dataelem.py

```diff
@@ -231,7 +231,12 @@ def from_json(
         vr: str,
         value: object,
         value_key: Union[str, None],
-        bulk_data_uri_handler: Optional[Callable[[str], object]] = None
+        bulk_data_uri_handler: Optional[
+            Union[
+                Callable[[BaseTag, str, str], object],
+                Callable[[str], object]
+            ]
+        ] = None
     ) -> _DataElement:
```

Comment:
So one thought: I'm thinking ahead to possible deprecation of the original signature, to simplify things long-term.  In that context, is it possibly better to change the handler signature by keeping the value first, and adding the tag and VR as optional parameters? Then people could fix their handler simply by adding the optional params as keywords defaulting to None).  And the typing is simpler with just:
```python
        bulk_data_uri_handler: Optional[
                Callable[[str, Optional[BaseTag], Optional[str]], object]
        ] = None
```
(I'm a 'typing' novice so perhaps that is not quite right.)

I'd admit though that there is not much difference because either way requires change in the user's handler.

Reply:
No worries. I had to research the 'typing' stuff for a hot minute to change what was there.

I made it tag-vr-value because that is how I (and probably others) think about data elements, but the parameter ordering isn't a hill I'm going to die on.

The optionality of the params really doesn't work for the pydicom code, as the calling function would then have to conditionally pass the last two parameters by examining the function signature, which is doable but felt kind of weird to me. It felt simpler and cleaner to wrap it on __init__.

I kind of like the ability for two signatures. For those that care, give the 3 param version. For those that don't, KISS. So my vote would be not to deprecate the original signature.

Reply:
> I made it tag-vr-value because that is how I (and probably others) think about data elements

Yes, I had thought about that too.


> I kind of like the ability for two signatures. For those that care, give the 3 param version. For those that don't, KISS. So my vote would be not to deprecate the original signature.

I'm okay with that; as I noted I'm not too worried about it.

I'll merge tomorrow after another quick look through.